tests/integration_tests/functional/test_cpu_template_helper.py

```diff
@@ -0,0 +1,264 @@
+                cpu_config_dict["cpuid"][
+                    (
+                        int(leaf_modifier["leaf"], 16),
+                        int(leaf_modifier["subleaf"], 16),
+                        register_modifier["register"],
+                    )
+                ] = int(register_modifier["bitmap"], 2)
+        # MSR
+        for msr_modifier in cpu_config_json["msr_modifiers"]:
+            cpu_config_dict["msrs"][int(msr_modifier["addr"], 16)] = int(
+                msr_modifier["bitmap"], 2
+            )
+
+    return cpu_config_dict
+
+
+CPUID_EXCEPTION_LIST = {
+    # CPUID.01h:ECX[OSXSAVE (bit 27)] is linked to CR4[OSXSAVE (bit 18)] that
+    # can be updated by guest OS.
+    # https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/arch/x86/kvm/x86.c?h=v5.10.176#n9872
+    (0x1, 0x0, "ecx"): 1 << 27,
+    # CPUID.07h:ECX[OSPKE (bit 4)] is linked to CR4[PKE (bit 22)] that can be
+    # updated by guest OS.
+    # https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/arch/x86/kvm/x86.c?h=v5.10.176#n9872
+    (0x7, 0x0, "ecx"): 1 << 4,
+    # CPUID.0Dh:EBX is variable depending on XCR0 that can be updated by guest
+    # OS with XSETBV instruction.
+    # https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/arch/x86/kvm/x86.c?h=v5.10.176#n973
+    (0xD, 0x0, "ebx"): 0xFFFFFFFF,
+    (0xD, 0x1, "ebx"): 0xFFFFFFFF,
+}
+
+
+def get_guest_cpuid(microvm, cpuid_list):
+    """
+    Return the guest MSR in the form of a dictionary where the key is a tuple
+    of (leaf, subleaf, register) and the value is the register value.
+    """
+    cpuid_dict = {}
+
+    for leaf, subleaf, _register in cpuid_list:
+        if (leaf, subleaf, "eax") in cpuid_dict:
+            continue
+
+        cpuid_cmd = f"cpuid -r -l {leaf} -s {subleaf} | head -n 2 | grep -v CPU"
+        code, stdout, stderr = microvm.ssh.execute_command(cpuid_cmd)
+        assert (
+            stderr.read() == ""
+        ), f"Failed to get CPUID for {leaf=:#x} / {subleaf=:#x}: {code=}"
+
+        # The output format is like follows:
+        #    0x00000001 0x00: eax=0x000606a6 ebx=0x00800800 ecx=0x7ffefbff edx=0xbfebfbff
+        results = stdout.read().split(": ")[1]
+        for result in results.split(" "):
+            splitted = result.split("=")
+            reg = splitted[0]
+            val = int(splitted[1], 16)
+            cpuid_dict[(leaf, subleaf, reg)] = val
+
+    return cpuid_dict
+
+
+MSR_EXCEPTION_LIST = [
+    # MSR_KVM_WALL_CLOCK and MSR_KVM_SYSTEM_TIME depend on the elapsed time.
+    0x11,
+    0x12,
+    # MSR_IA32_FEAT_CTL and MSR_IA32_SPEC_CTRL are R/W MSRs that can be
+    # modified by OS to control features.
+    0x3A,
+    0x48,
+    # MSR_IA32_SMBASE is not accessible outside of System Management Mode.
+    0x9E,
+    # MSR_IA32_ARCH_CAPABILITIES is not available on AMD. Firecracker
+    # disables it explicitly by set 0 to CPUID.(EAX=07H,ECX=0):EDX[bit 29].
+    0x10A,
```

Comment:
Can we include it on Intel? It might be a good candidate for bit flipping due to microcode update.

Reply:
Super good point. I believe we should. will fix.

And I also found 0x10a exist in the AMD Milan + host 5.10 result: https://github.com/firecracker-microvm/firecracker/pull/3586/files#diff-39cc840cc5c3c8060a43b301205e18c040576804c01a13ccbd74ee5bd2b9ff09R1390-R1393
Will investigate this as well.

Reply:
0x10a MSR is listed on AMD Milan + kernel 5.10, because kernel 5.7 or later supports software emulation of this MSR and KVM_GET_MSR_INDEX_LIST is a system ioctl and doesn't care CPUID setting that firecracker does in the normalization. Thus, I removed this MSR only on AMD from the list of dumpable MSRs and enabled this MSR check in the test.

Reply:
done.